internal/jstest/jstest.go

```diff
@@ -141,20 +142,56 @@ func Run(ctx context.Context, dir string, args []string) (*internal.TestResults,
 	return res, nil
 }
 
-// runCommand runs command with args inside the mongo container and returns the
-// combined output.
-func runCommand(dir, command string, args ...string) ([]byte, error) {
+// runShellWithScript runs the mongo shell inside a container with script and returns the combined output.
+func runShellWithScript(dir, script string) ([]byte, error) {
 	bin, err := exec.LookPath("docker")
 	if err != nil {
 		return nil, err
 	}
 
-	args = append([]string{"--verbose", "--norc", "mongodb://host.docker.internal:27017/"}, args...)
-	dockerArgs := append([]string{"compose", "run", "-T", "--rm", command}, args...)
+	dockerArgs := []string{"compose", "run", "-T", "--rm", "mongo"}
+	shellArgs := []string{
+		"--verbose", "--norc", "mongodb://host.docker.internal:27017/",
+		"--eval", evalBuilder(script, nil), script,
+	}
+	dockerArgs = append(dockerArgs, shellArgs...)
+
 	cmd := exec.Command(bin, dockerArgs...)
 	cmd.Dir = dir
 
-	log.Printf("Running %s", strings.Join(cmd.Args, " "))
+	log.Printf("Running %s", script)
 
 	return cmd.CombinedOutput()
 }
+
+// evalBuilder creates the TestData object and sets the testName property for the shell.
+// It optionally sets additional properties on the TestData object.
+func evalBuilder(script string, obj map[string]string) string {
+	var eb bytes.Buffer
+	eb.WriteString("TestData = new Object();")
+	eb.WriteByte(' ')
```

Comment:
```go
fmt.Fprint(&eb, "TestData = new Object(); ")
```

I think we could simplify it a bit by using only one family of functions instead of both `fmt.FprintXXX` and `eb.WriteXXX`